### PR TITLE
[TECHNICAL SUPPORT] LPS-184947 Throw a new NoSuchLayoutException in case we are using a c…

### PR DESCRIPTION
**Pull request by NemethNorbert**
Hey,

Previous PR: https://github.com/liferay-lima/liferay-portal/pull/4320
[LPS-184947](https://issues.liferay.com/browse/LPS-184947)

I noticed in case we are using a custom no page found url, the fix does not work. I made an additional commit to resolve that as well. Please review my change.

Thanks,

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:relevant

File: modules/apps/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/internal/servlet/FriendlyURLServlet.java [hunk 1]
```diff
@@ -393,7 +393,11 @@ public Redirect getRedirect(
 				if (Validator.isNotNull(
 						PropsValues.LAYOUT_FRIENDLY_URL_PAGE_NOT_FOUND)) {
 
-					throw exception;
+					if (exception instanceof NoSuchLayoutException) {
+						throw exception;
+					}
+
+					throw new NoSuchLayoutException();
```
Comment (adolfopa) [inline modules/apps/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/internal/servlet/FriendlyURLServlet.java:400]: Can you wrap the original exception in the new `NoSuchLayoutException`?

**Review by adolfopa — CHANGES_REQUESTED** [review]

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:forward

**Comment by adolfopa** [comment]
ci:forward

**Comment by Hanlf** [comment]
ci:test:relevant

**Comment by Hanlf** [comment]
The above failure is caused by [LPS-186487](https://issues.liferay.com/browse/LPS-186487), not related to this PR.

**Comment by Hanlf** [comment]
ci:forward:force

**Pull request closed without merging**